### PR TITLE
Store final transcripts locally as .txt files

### PR DESCRIPTION
**Pull request by nikvaessen**

### CODE REVIEW

File: src/main/java/org/jitsi/jigasi/transcription/Transcriber.java [hunk 1]
```diff
@@ -127,30 +126,52 @@
...
+    public Transcriber(String roomName, TranscriptionService service)
     {
         if(!service.supportsStreamRecognition())
         {
             throw new IllegalArgumentException("Currently only services which" +
-                    " support streaming recognition are supported");
+                " support streaming recognition are supported");
         }
         this.transcriptionService = service;
+        addTranscriptionListener(this.transcript);
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/Transcriber.java]: Should we also handle roomName?

File: src/main/java/org/jitsi/jigasi/transcription/Transcriber.java [hunk 2]
```diff
@@ -192,8 +217,13 @@ public void start()
...
         if(State.NOT_STARTED.equals(this.state))
         {
+            logger.debug("Transcriber is now transcribing");
             this.state = State.TRANSCRIBING;
-            executorService = Executors.newSingleThreadExecutor();
+            this.executorService = Executors.newSingleThreadExecutor();
+
+            List<String> names = new ArrayList<>(participants.size());
+            participants.values().forEach((p) -> names.add(p.getName()));
+            this.transcript.started(roomName, names);
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/Transcriber.java]: When we list the participants, the ones without a name should be listed with IDs. Perhaps we can handle this by modifying Participant#getName, or handle it here?

File: src/main/java/org/jitsi/jigasi/transcription/Transcriber.java [hunk 3]
```diff
@@ -336,6 +371,30 @@ public AudioMixerMediaDevice getMediaDevice()
     }
 
     /**
+     * If all participant have been completely transcribed, and state is
+     * FINISHING_UP, the state can be set to FINISHED
+     */
+    private void isFinishedUp()
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/Transcriber.java]: This is confusing -- the name starts with "is", but it doesn't return a boolean? Can you please rename and clarify the doc?

File: src/main/java/org/jitsi/jigasi/transcription/Transcript.java [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+    implements TranscriptionListener
+{
+    /**
+     * Formats an Instant to <hour:minute:second> in the timezone the machine
+     * is running on
+     */
+    private static final DateTimeFormatter timeFormatter
+        = DateTimeFormatter
+        .ofLocalizedTime(FormatStyle.MEDIUM)
+        .withZone(ZoneId.systemDefault());
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/Transcript.java]: I think we should either use UTC, or explicitly include the timezone somewhere. What do you think?

File: src/main/java/org/jitsi/jigasi/transcription/Transcript.java [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+
+    /**
+     * Notify the transcript that the conference started at this exact moment
+     * Will be ignored if already told the conference started
+     *
+     * @param initialNames the names of the participants currently in the room
+     */
+    public void started(List<String> initialNames)
+    {
+        if(started != null)
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/Transcript.java]: Shouldn't this be started==null?

File: src/main/java/org/jitsi/jigasi/transcription/LocalTxtTranscriptHandler.java [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+    /**
+     * The logger of this class
+     */
+    private static final Logger logger = Logger.getLogger(Transcript.class);
+
+    /**
+     * The root dir where a new directory will be created in to store
+     * transcripts
+     */
+    private static final String ROOT_DIR = System.getProperty("user.dir");
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/LocalTxtTranscriptHandler.java]: Can you use the config service and a separate property name, e.g. org.jitsi.jigasi.transcription.LocalTxtTranscriptHandler.DIRECTORY (which can combine ROOT_DIR and LOG_DIR)?

File: src/main/java/org/jitsi/jigasi/transcription/LocalTxtTranscriptHandler.java [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+                return;
+            }
+        }
+
+        File t = new File(logDir, "transcript_" + Instant.now() + ".txt");
+
+        try(FileWriter writer = new FileWriter(t))
+        {
+            writer.write(transcript);
+            System.out.println(transcript);
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/LocalTxtTranscriptHandler.java]: Can we just print one line that acknowledges that we saved it in a file?

File: src/main/java/org/jitsi/jigasi/transcription/Transcriber.java (hunk 1 above)
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/Transcriber.java]: fixed

File: src/main/java/org/jitsi/jigasi/transcription/Transcriber.java (hunk 2 above)
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/Transcriber.java]: fixed

File: src/main/java/org/jitsi/jigasi/transcription/Transcriber.java (hunk 3 above)
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/Transcriber.java]: fixed

File: src/main/java/org/jitsi/jigasi/transcription/Transcript.java (hunk 5 above)
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/Transcript.java]: fixed

File: src/main/java/org/jitsi/jigasi/transcription/LocalTxtTranscriptHandler.java (hunk 6 above)
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/LocalTxtTranscriptHandler.java]: org.jitsi.jigasi.transcription.LocalTxtTranscriptHandler.DIRECTORY can now be added to sip-communicator.properties

File: src/main/java/org/jitsi/jigasi/transcription/LocalTxtTranscriptHandler.java (hunk 7 above)
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/LocalTxtTranscriptHandler.java]: fixed

File: src/main/java/org/jitsi/jigasi/transcription/Transcript.java (hunk 4 above)
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/Transcript.java]: we decided on UTC in meetings if I recall correctly, so set it to that

**Pull request merged**